File: src/main/java/onl/tesseract/entityConverter/DurationConverter.java
package onl.tesseract.entityConverter;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.time.Duration;
@Converter(autoApply = true)
public class DurationConverter implements AttributeConverter<Duration, Long> {

    @Override
    public Long convertToDatabaseColumn(Duration duration) {
        if(duration == null) {
            return null;
        }
        return duration.toSeconds();
    }

    @Override
    public Duration convertToEntityAttribute(Long l) {
        if (l == null){
            return null;
        }
        return Duration.ofSeconds(l);
    }
}

File: src/main/java/onl/tesseract/entityConverter/DurationConverter.kt
package onl.tesseract.entityConverter

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.time.Duration

@Converter(autoApply = true)
class DurationConverter : AttributeConverter<Duration?, Long?> {
    override fun convertToDatabaseColumn(duration: Duration?): Long? {
        if (duration == null) {
            return null
        }
        print(duration.toSeconds())
        return duration.toSeconds()
    }

    override fun convertToEntityAttribute(l: Long?): Duration? {
        if (l == null) {
            return null
        }
        return Duration.ofSeconds(l)
    }
}
